os/kernel/debug/memdbg.c

```diff
@@ -0,0 +1,100 @@
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <debug.h>
+
+#include <tinyara/arch.h>
+#include <tinyara/mm/mm.h>
+
+/***************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+
+extern const uint32_t g_idle_topstack;
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: display_memory_information
+ *
+ * Description:
+ *   This function prints the memory information for debug such as
+ *   the RAM configuration, Idle stack configuration, Heap configuration.
+ *
+ * Inputs:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *   None
+ *
+ ****************************************************************************/
+
+void display_memory_information(void)
+{
+	/* Print RAM configuration */
+
+	mllvdbg("-------------------RAM Configuration--------------------\n");
+#ifdef CONFIG_SRAM_START_ADDR
+	mllvdbg("Internal RAM   : start address = 0x%x size = %u \n", CONFIG_SRAM_START_ADDR, (CONFIG_KSRAM_SIZE + CONFIG_USRAM_SIZE));
+#if (defined(CONFIG_KSRAM_SIZE) && defined(CONFIG_RAM_KREGIONx_START))
+	mllvdbg("Kernel RAM     : start address = 0x%x size = %u \n", kregionx_start[0], CONFIG_KSRAM_SIZE);
+#endif
+#if (defined(CONFIG_USRAM_SIZE) && defined(CONFIG_RAM_REGIONx_START))
+	mllvdbg("User RAM       : start address = 0x%x size = %u \n", regionx_start[0], CONFIG_USRAM_SIZE);
+#endif
+#endif
+
+	/* Print Idle Stack configuration */
+
+	mllvdbg("----------------Idle Stack Configuration----------------\n");
+	mllvdbg("Idle Stack     : base address = 0x%x size = %u \n", g_idle_topstack, CONFIG_IDLETHREAD_STACKSIZE);
```

Comment:
To reduce RO data and to reduce uart operation,
let's remove "Idle stack" from this line. Because you already write that at previous line.

Reply:
done